### PR TITLE
fix problem if no initialValues are served to FinalFormRangeInput, and adjusted base styling

### PR DESCRIPTION
**Pull request by MFSepplive**

### CODE REVIEW

**Review by johnnyomair — APPROVED** [review]

**Review by jamesricky — APPROVED** [review]

**Review by kaufmo — APPROVED** [review]

**Comment by nsams** [comment]
could you please add a development story where this can be tested? (I'm pretty sure you created/adjusted one to test this, so please commit it)

**Comment by MFSepplive** [comment]
> could you please add a development story where this can be tested? (I'm pretty sure you created/adjusted one to test this, so please commit it)

i used the same story i used before the fix :) i just tried to remove the initialValues to try to see if it works :) i didnt make another story for showing the fix :)

**Comment by nsams** [comment]
>  i didnt make another story for showing the fix :)

Ok. And I'm asking you to create a new story and add it to this PR.

I'd like to test and discuss your change, as it's not completely clear if this is the desired behaviour. And that is soo much easier if we can do that thru the netlify deploy instead of starting it locally and modifying the story. And it's a general thing we want to do: create development stories, also for reproducing bugs.

**Comment by MFSepplive** [comment]
> > i didnt make another story for showing the fix :)
> 
> Ok. And I'm asking you to create a new story and add it to this PR.
> 
> I'd like to test and discuss your change, as it's not completely clear if this is the desired behaviour. And that is soo much easier if we can do that thru the netlify deploy instead of starting it locally and modifying the story. And it's a general thing we want to do: create development stories, also for reproducing bugs.

okay i will create a Range Slider Story with no initialValues to show how it works without the initialValues :) or what kind of a story would you like to see ?

**Comment by MFSepplive** [comment]
> > i didnt make another story for showing the fix :)
> 
> Ok. And I'm asking you to create a new story and add it to this PR.
> 
> I'd like to test and discuss your change, as it's not completely clear if this is the desired behaviour. And that is soo much easier if we can do that thru the netlify deploy instead of starting it locally and modifying the story. And it's a general thing we want to do: create development stories, also for reproducing bugs.

i added 2 different variants on how the finalFormRangeInput can work to the range Input story

**Comment by nsams** [comment]
as disscussed yesterday: if the field has no value, use empty value for inputs. Min/Max should be shown as placeholder in the inputs.

**Comment by MFSepplive** [comment]
> as disscussed yesterday: if the field has no value, use empty value for inputs. Min/Max should be shown as placeholder in the inputs.

i changed the behaviour of the FinalFormRangeInput so it shows placeholder text if no initialValues are set.

File: packages/admin/src/form/FinalFormRangeInput.tsx [hunk 1]
```diff
@@ -74,21 +78,21 @@ const FinalFormRangeInputComponent: React.FunctionComponent<WithStyles<typeof st
                         <InputBase
                             name={`${name}.min`}
                             inputProps={{
-                                min: min,
-                                max: fieldValue.max,
-                                value: internalMinInput,
-                                type: "number",
+                                value: internalMinInput !== undefined ? internalMinInput : "",
+                                type: "text",
```
Comment (nsams) [inline packages/admin/src/form/FinalFormRangeInput.tsx:82]: why did you change that to text?

File: packages/admin/src/form/FinalFormRangeInput.tsx [hunk 2]
```diff
@@ -74,21 +78,21 @@ const FinalFormRangeInputComponent: React.FunctionComponent<WithStyles<typeof st
...
                             startAdornment={startAdornment}
                             endAdornment={endAdornment}
                             onBlur={() => {
-                                const minFieldValue = Math.min(internalMinInput, fieldValue.max);
-                                if (fieldValue.min !== minFieldValue) {
-                                    onChange({ ...fieldValue, min: minFieldValue < min ? min : minFieldValue });
-                                }
+                                const minFieldValue = Math.min(internalMinInput ? internalMinInput : min, fieldValue.max ? fieldValue.max : max);
+
+                                onChange({ ...fieldValue, min: minFieldValue < min ? min : minFieldValue });
```
Comment (nsams) [inline packages/admin/src/form/FinalFormRangeInput.tsx:90]: Now when I focus the field, do nothing and blur it you write max into it. That doesn't feel right.

And shouldn't (now that we support undefined value) "" input value set it back to undefined?

File: packages/admin-stories/src/admin/form/RangeInput.tsx [hunk 3]
```diff
@@ -37,24 +38,109 @@ const Thumb = styled.div`
...
+                                endAdornment={<span>€</span>}
+                                sliderProps={{ ThumbComponent: Thumb }}
+                            />
+                            <Button
+                                variant="contained"
+                                color="primary"
+                                onClick={() => {
+                                    for (const name in values) {
+                                        //required because reset removes values
+                                        form.change(name, { min: undefined, max: undefined });
```
Comment (nsams) [inline packages/admin-stories/src/admin/form/RangeInput.tsx:99]: reset should be supported - what is the problem with (completely) removed value?

File: packages/admin/src/form/FinalFormRangeInput.tsx (hunk 1 above)
Comment (MFSepplive) [reply packages/admin/src/form/FinalFormRangeInput.tsx:82]: because if it is number you cant remove the value :) undefined is only in a text field possible :)

File: packages/admin/src/form/FinalFormRangeInput.tsx (hunk 2 above)
Comment (MFSepplive) [reply packages/admin/src/form/FinalFormRangeInput.tsx:90]: ok i thought that would be ok, i will change that asap

File: packages/admin-stories/src/admin/form/RangeInput.tsx (hunk 3 above)
Comment (MFSepplive) [reply packages/admin-stories/src/admin/form/RangeInput.tsx:99]: thats true :) forgot about that :) i changed it to form.reset() :)

File: packages/admin/src/form/FinalFormRangeInput.tsx (hunk 1 above)
Comment (nsams) [reply packages/admin/src/form/FinalFormRangeInput.tsx:82]: You can remove the value of a type=text input. Please change that back.

try here: https://jsfiddle.net/nu8cf6k9/1/

**Comment by nsams** [comment]
looks good!

one thing I noticed:
- go to the without initialValues story
- enter a value in the max input
- notice the min stays empty
- click reset
- now drag the max dot to enter a value
- notice min is now set to "0"

that's not completely consistent, what do you think?

**Comment by MFSepplive** [comment]
> * ow drag the max dot

i dont know how to approach that other then that, because the MUI Slider cant handle undefined values, so if you drag the dot i have to set min and max Value because they have to be sync (Input fields and slider). I know its a bit inconsistent but it prevents unexpected errors with undefined values. 
I think if you drag on the slider then your intention is to change the value so it is ok to set the missing Value too :) but thats just my opinion on that  :)

**Comment by nsams** [comment]
Or the other way around: on max blur you also set min input to min.

Actually, the field value type is `{ min: number; max: number }`, so `{ min: undefined, max: 150 }` is invalid.

**Comment by MFSepplive** [comment]
> Or the other way around: on max blur you also set min input to min.
> 
> Actually, the field value type is `{ min: number; max: number }`, so `{ min: undefined, max: 150 }` is invalid.

so if i set for example the min Value and the max Value is undefined, how should the form field behave ? set max to maxSliderValue or should it stay undefined and just change the type of the fieldRenderProps ? 
or how would think should it behave ?

**Comment by nsams** [comment]
> so if i set for example the min Value and the max Value is undefined, how should the form field behave ? set max to maxSliderValue

yes, that also was my first thought.

> or should it stay undefined and just change the type of the fieldRenderProps ?

could be done, but that would make it harder to use the value.

**Comment by MFSepplive** [comment]
> > so if i set for example the min Value and the max Value is undefined, how should the form field behave ? set max to maxSliderValue
> 
> yes, that also was my first thought.
> 
> > or should it stay undefined and just change the type of the fieldRenderProps ?
> 
> could be done, but that would make it harder to use the value.

changed it :)

**Comment by nsams** [comment]
now we have another bug:
- open "without initialValues" story
- enter 100 in max input
- blur
- min gets set to "0"
- remove 100 from max input (using backspace)
- blur
- notice how max is "" and min in "0" (expected: unsure, but probably min should be "")

**Comment by MFSepplive** [comment]
> now we have another bug:
> 
> * open "without initialValues" story
> * enter 100 in max input
> * blur
> * min gets set to "0"
> * remove 100 from max input (using backspace)
> * blur
> * notice how max is "" and min in "0" (expected: unsure, but probably min should be "")

so if 1 Value (either min or max) is undefined/"", the other value should be the same then ?

**Comment by nsams** [comment]
yes; the value of FinalFormRangeInput can be
- either undefined (which results in min+max undefined)
- or `{ min: number, max: number }`

and you have to make sure the UI never shows a state that would result in `{ min: undefined, max: number }` or `{ min: number, max: undefined }`

**Comment by MFSepplive** [comment]
> yes; the value of FinalFormRangeInput can be
> 
> * either undefined (which results in min+max undefined)
> * or `{ min: number, max: number }`
> 
> and you have to make sure the UI never shows a state that would result in `{ min: undefined, max: number }` or `{ min: number, max: undefined }`

now only values like `{min: undefined, max: undefined}` or `{min: number, max: number}` are possible :)

**Comment by nsams** [comment]
The value has to be `undefined`. A value of `{min: undefined, max: undefined}` is not compatible with the type `{min: number, max: number}`

For easier verifying this, please output the form values in the story (similar to https://final-form.org/docs/react-final-form/examples/simple)

**Review by nsams — APPROVED** [review]
